tools/docker/Dockerfile

```diff
@@ -34,50 +43,38 @@ ENV LANG en_US.UTF-8
 
 # -- Android SDK ------------------------------------------------------------------------
 
-RUN cd /opt && wget -q https://dl.google.com/android/android-sdk_r24.4.1-linux.tgz -O android-sdk.tgz
-RUN cd /opt && tar -xvzf android-sdk.tgz
-RUN cd /opt && rm -f android-sdk.tgz
-
-ENV ANDROID_SDK_HOME /opt/android-sdk-linux
-ENV ANDROID_HOME /opt/android-sdk-linux
-ENV PATH ${PATH}:${ANDROID_SDK_HOME}/tools:${ANDROID_SDK_HOME}/platform-tools:/opt/tools
-
-# Platform tools
-RUN echo y | android update sdk --no-ui --all --filter platform-tools | grep 'package installed'
-
-# Android SDK
-RUN echo y | android update sdk --no-ui --all --filter android-26 | grep 'package installed'
-
-# Build tools
-RUN echo y | android update sdk --no-ui --all --filter build-tools-26.0.2 | grep 'package installed'
-RUN echo y | android update sdk --no-ui --all --filter build-tools-27.0.3 | grep 'package installed'
+RUN mkdir -p /build/android-sdk
+WORKDIR /build
 
-# Extras
-RUN echo y | android update sdk --no-ui --all --filter extra-android-m2repository | grep 'package installed'
-RUN echo y | android update sdk --no-ui --all --filter extra-google-m2repository | grep 'package installed'
-RUN echo y | android update sdk --no-ui --all --filter extra-google-google_play_services | grep 'package installed'
+ENV ANDROID_HOME /build/android-sdk
+ENV ANDROID_SDK_HOME /build/android-sdk
+ENV PATH ${PATH}:${ANDROID_SDK_HOME}/tools:${ANDROID_SDK_HOME}/tools/bin:${ANDROID_SDK_HOME}/platform-tools:/opt/tools:${ANDROID_SDK_HOME}/build-tools/${ANDROID_BUILD_TOOLS}
 
-# Copy tools from our repository into the docker image
-COPY tools /opt/tools
+RUN curl -L https://dl.google.com/android/repository/sdk-tools-linux-${ANDROID_SDK_VERSION}.zip > sdk.zip \
+        && unzip sdk.zip -d ${ANDROID_SDK_HOME} \
+        && rm sdk.zip
 
-# Accept licenses
-COPY licenses ${ANDROID_SDK_HOME}/licenses
+RUN mkdir -p /build/android-sdk/.android/
+RUN touch /build/android-sdk/.android/repositories.cfg
 
-# -- Update SDK -------------------------------------------------------------------------
+RUN yes | sdkmanager --licenses
 
-# Update SDK
-RUN /opt/tools/android-accept-licenses.sh android update sdk --no-ui --obsolete --force
+RUN sdkmanager --verbose "platform-tools" \
+    "platforms;android-${ANDROID_PLATFORM_VERSION}" \
+    "build-tools;${ANDROID_BUILD_TOOLS}" \
+    "extras;android;m2repository" \
+    "extras;google;m2repository"
 
 # -- Project setup ----------------------------------------------------------------------
 
 WORKDIR /opt
 
 # Checkout source code
-RUN git clone https://github.com/mozilla-mobile/firefox-tv.git
+RUN git clone ${PROJECT_REPOSITORY}
 
 # Build project and run gradle tasks once to pull all dependencies
 WORKDIR /opt/firefox-tv/
-RUN ./gradlew clean assemble lint checkstyle ktlint pmd test
+RUN ./gradlew --no-daemon clean assemble lint checkstyle ktlint pmd detektCheck
```

Comment:
We don't want to run the unit tests too?

Reply:
I'm running into an issue where gradle gets killed (exit code 137) during test run - but only when building the Docker image. Running the tests in the built image works without any issues though. I'll file a follow-up issue. I think we have fixed a similar issue in Focus some time ago.

Reply:
#980